src/main/java/org/opensearch/jobscheduler/transport/GetJobTypeTransportAction.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.jobscheduler.transport;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.HandledTransportAction;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.rest.RestStatus;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportService;
+
+import static org.opensearch.jobscheduler.utils.RestHandlerUtils.wrapRestActionListener;
+
+/**
+ * Get Job Type Transport Action
+ */
+public class GetJobTypeTransportAction extends HandledTransportAction<GetJobTypeRequest, GetJobDetailsResponse> {
+    private static final Logger LOG = LogManager.getLogger(GetJobTypeTransportAction.class);
+
+    @Inject
+    public GetJobTypeTransportAction(String actionName, TransportService transportService, ActionFilters actionFilters) {
+        super(GetJobTypeAction.NAME, transportService, actionFilters, GetJobTypeRequest::new);
+    }
+
+    @Override
+    protected void doExecute(Task task, GetJobTypeRequest request, ActionListener<GetJobDetailsResponse> actionListener) {
```

Comment:
What are we really processing in the transport layer? 
I see pretty much nothing we can't do at the Rest Layer.

Reply:
To return the response as 
{
"response":"success"
}

we need to implement the above logic in transport as layer. As the prepare request method return type is RestChannelConsumer, therefore we need to implement the logic for the same in transport layer. The response cannot be returned from Rest layer only

Reply:
I understand we have to respond with a success message, but what are we processing the transport layer from the request params? Pretty much nothing. 

We could just send the response from Rest `prepareRequest` as:

```
return new RestChannelConsumer() {
            
            final String mode = request.param("mode","strict");

            @Override
            public void accept(RestChannel channel) throws Exception {
                RestStatus restStatus = RestStatus.OK;
                BytesRestResponse response = null;
                try {
                    String status = "Sucess";
                    String message = "OK";
                    builder.startObject();

    
                    builder.field("message", message);
                    builder.field("mode", mode);
                    builder.field("status", status);
                    builder.endObject();
                    response = new BytesRestResponse(restStatus, builder);
                } finally {
                    builder.close();
                }
                channel.sendResponse(response);
            }
        };
```

Reply:
Implemented the above approach in the latest PR